docs/ref/models/database-functions.txt

```diff
@@ -293,3 +293,518 @@ Usage example::
+Returns the hour of ``date_expression``.
+
+Usage example::
+
+    >>> from datetime import datetime
+    >>> from django.db.models.functions import ExtractHour
+    >>> start = datetime(2015, 6, 15, 14)
+    >>> Experiment.objects.create(start_time=start)
+    >>> experiment = Experiment.objects.annotate(
+    ...    start_hour=ExtractHour('start_time')).get()
+    >>> print(experiment.start_hour)
+    14
+    >>> Experiment.objects.filter(
+    ...    start_time__hour=ExtractHour('start_time')).count()
+    1
+
+``ExtractMinute``
+-----------------
+
+.. class:: ExtractMinute(date_expression, **extra)
+
+    .. attribute:: lookup_name = 'minute'
+
+Logically equivalent to ``Extract('date_field', 'minute')``. ``ExtractMinute``
+is also a ``Transform`` registered on Date and DateTime fields as ``__minute``.
+Returns the minute of ``date_expression``.
+
+Usage example::
+
+    >>> from datetime import datetime
+    >>> from django.db.models.functions import ExtractMinute
+    >>> start = datetime(2015, 6, 15, 14, 30)
+    >>> Experiment.objects.create(start_time=start)
+    >>> experiment = Experiment.objects.annotate(
+    ...    start_minute=ExtractMinute('start_time')).get()
+    >>> print(experiment.start_minute)
+    30
+    >>> Experiment.objects.filter(
+    ...    start_time__minute=ExtractMinute('start_time')).count()
+    1
+
+``ExtractSecond``
+-----------------
+
+.. class:: ExtractSecond(date_expression, **extra)
+
+    .. attribute:: lookup_name = 'second'
+
+Logically equivalent to ``Extract('date_field', 'second')``. ``ExtractSecond``
+is also a ``Transform`` registered on Date and DateTime fields as ``__second``.
+Returns the second of ``date_expression``.
+
+Usage example::
+
+    >>> from datetime import datetime
+    >>> from django.db.models.functions import ExtractSecond
+    >>> start = datetime(2015, 6, 15, 14, 30, 50)
+    >>> Experiment.objects.create(start_time=start)
+    >>> experiment = Experiment.objects.annotate(
+    ...    start_second=ExtractSecond('start_time')).get()
+    >>> print(experiment.start_second)
+    50
+    >>> Experiment.objects.filter(
+    ...    start_time__second=ExtractSecond('start_time')).count()
+    1
+
+``Trunc``
+---------
+
+.. class:: Trunc(date_expression, kind, output_field=None, **extra)
+
+Truncates a date up to a significant component.
+
+When you only care if something happened in a particular year, hour, or day,
+but not the exact moment in time, then ``Trunc`` (and its subclasses) can be
+useful to filter or group your data. Calculating the number of sales per day
+is a common use case for ``Trunc``.
+
+``Trunc`` takes a single ``date_expression`` representing a ``DateField`` or
+``DateTimeField``, a ``kind`` representing a date part, and an ``output_field``
+that's either ``DateTimeField()`` or ``DateField()``. It returns a datetime or
+date, depending on ``output_field``, with fields up to ``kind`` set to their
+minimum value.
```

Comment:
``output_field`` should default to the type of the original field.

It isn't clear to me that truncating aware datetimes and getting dates as a result can be a valid operation. I think you must return always return aware datetimes in that case.